[FIPS] build operator in fips mode

SeanMooney opened the pull request:
This change modifies how the operator is built to use
the ubi9 base image and golang tool chain.

A new GO_BUILD_EXTRA_ENV_ARGS build arg is added to the dockerfile
to enable the build system to inject addtional parmaters.
the default value enable dynmaic linking of glibc which is required
for FIPS

The prow_ci.env file is extended to enable fips checkign in the
prow image build job to validate this change.

The Makefile is extened to allow passing build args to docker-build

Closes: [OSPRH-3110](https://issues.redhat.com//browse/OSPRH-3110)

mrKisaoLamb commented:
/retest

gibizer reviewed (approved):
Looks good. It runs the new linters now:
```
level=info msg="[lintersdb] Active 11 linters: [errcheck errorlint ginkgolinter gofmt gosimple govet ineffassign revive staticcheck typecheck unused]"
```

SeanMooney committed:
[FIPS] build operator in fips mode

This change modifies how the operator is built to use
the ubi9 base image and golang tool chain.

A new GO_BUILD_EXTRA_ENV_ARGS build arg is added to the dockerfile
to enable the build system to inject addtional parmaters.
the default value enable dynmaic linking of glibc which is required
for FIPS

The prow_ci.env file is extended to enable fips checkign in the
prow image build job to validate this change.

The Makefile is extened to allow passing build args to docker-build

gibizer commented:
/test placement-operator-build-deploy-kuttl
Something is wrong with the build. Let see if it is persistent
```
++ /bin/bash hack/pin-bundle-images.sh
parse error: Invalid numeric literal at line 1, column 10
+ OPENSTACK_BUNDLES=,quay.io/openstack-k8s-operators/barbican-operator-bundle:655a181feae028aae211071a235a66bacb1ef402,quay.io/openstack-k8s-operators/cinder-operator-bundle:5e8e748d6046beee57bf3e4271565ff3bb7f8d11,quay.io/openstack-k8s-operators/dataplane-operator-bundle:eb350187c54563d7d0035c78450d8a320ca34dcb,quay.io/openstack-k8s-operators/designate-operator-bundle:620a93388acfb9b46d416d695a19dda1f0f99874,quay.io/openstack-k8s-operators/glance-operator-bundle:d3646fda95358c60fe8c82a4b9102d2638702b36,quay.io/openstack-k8s-operators/heat-operator-bundle:35cd4930ad3be84ee5902db7fa108479ebc7162e,quay.io/openstack-k8s-operators/horizon-operator-bundle:ddc6aa0dcf47dc6d89dacd83c2b1517abc6de67a,quay.io/openstack-k8s-operators/infra-operator-bundle:e4efedde5776adb91ed98cc96d9cfab53227e5f5,quay.io/openstack-k8s-operators/ironic-operator-bundle:8b6a4ca3bdc598c51898419b034764fa5cfc21ec,quay.io/openstack-k8s-operators/keystone-operator-bundle:34883c60812b94aa36e9997219eb2ecfafbbec99,quay.io/openstack-k8s-operators/manila-operator-bundle:5a5000e56580b0d1b005742fdae61bb3fd6f38b1,quay.io/openstack-k8s-operators/mariadb-operator-bundle:3dcb5d5b24f79c075b2e142538256e61361f2d33,quay.io/openstack-k8s-operators/neutron-operator-bundle:ca38cd1c0fd78cbe006ed492a643edb6f6ee4940,quay.io/openstack-k8s-operators/nova-operator-bundle:0a39e8ee1c0724e59e0d44e37474734e2c35b9ad,quay.io/openstack-k8s-operators/octavia-operator-bundle:fca054830e064de5f3e97b9a7696747d83b476e9,quay.io/openstack-k8s-operators/openstack-ansibleee-operator-bundle:77a33066c0959daf541d9bb6fccdb20a83c85552,quay.io/openstack-k8s-operators/openstack-baremetal-operator-bundle:f184903f87062eee55a7f71ed9d3366862b496e0,quay.io/openstack-k8s-operators/ovn-operator-bundle:41e2d7f8870e5e64802685ef6ece5f6b9cce3acc
```

gibizer commented:
yepp same build failure again, it is persistent

SeanMooney commented:
/test placement-operator-build-deploy-kuttl

rlandy commented:
/test placement-operator-build-deploy-kuttl

viroel commented:
As discussed pointed in chat earlier, _pin-bundle-images.sh_ is failing because the fork has a different repo name, and the script's regex does not expect that.

SeanMooney commented:
oh right i have 2 forks ...

the first one https://github.com/SeanMooney/placement-operator was one i created whne the placement operator was acidnetally deleted i created it before i setup my gitea mirrors
https://gitea.seanmooney.info/openstack-k8s-operators/placement-operator

i can delete my first fork and then rename the new one since that is the one git thinks is the correct fork.

SeanMooney commented:
/test placement-operator-build-deploy-kuttl
